pkg/component/shared/otelcollector.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package shared
+
+import (
+	"strconv"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/imagevector"
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/component/observability/logging/vali/constants"
+	"github.com/gardener/gardener/pkg/component/observability/opentelemetry/collector"
+)
+
+// NewOtelCollector returns new OtelCollector deployer
+func NewOtelCollector(
+	c client.Client,
+	namespace string,
+) (
+	component.DeployWaiter,
+	error,
+) {
+	image, err := imagevector.Containers().FindImage(imagevector.ContainerImageNameOpentelemetryCollectorControlPlane)
+	if err != nil {
+		return nil, err
+	}
+
+	deployer := collector.New(
+		c,
+		namespace,
+		collector.Values{
+			Image: image.String(),
+		},
+		"http://"+constants.ServiceName+":"+strconv.Itoa(constants.ValiPort)+constants.PushEndpoint,
```

Comment:
Could we utilize something like this?
```suggestion
		path.Join("http://"+constants.ServiceName+":"+strconv.Itoa(constants.ValiPort), constants.PushEndpoint),
```

Reply:
Are you suggesting this so that the final url does not depend on whether the `PushEndpoint` constant starts with a slash or not?

Reply:
Essentially, yes.
When I was just looking at that, I was a bit confused.
Maybe this makes it more unreadable within this URL building 😅

Reply:
I don't prefer either when it comes to readability, but I'll add your suggestion because of the mentioned benefit

Reply:
Maybe someone else has a stronger opinion on that 😅
But I think you can add it for now.